### PR TITLE
#9447:Overload helpers so Symbol is not required

### PR DESCRIPTION
**Pull request by cookiee01**
Overloaded helpers to accept `String`

### CODE REVIEW

**Comment by cookiee01** [comment]
Binary compatibility failure seems to be caused by change in constructor type of `FieldElements`. I am not sure if we can avoid this. Would really appreciate some help here.

Thanks !

**Comment by dwijnand** [comment]
Hey @cookiee01.  Out use of MiMa on the master branch is only for informative purposes.  So you can add an exclusion for this.

**Comment by ignasi35** [comment]
Just add 

```
// Overload helpers so Symbol is not required
ProblemFilters.exclude[MissingTypesProblem]("views.html.helper.FieldElements$")
```

in https://github.com/playframework/playframework/blob/master/project/BuildSettings.scala#L528

File: project/BuildSettings.scala [hunk 1]
```diff
@@ -515,6 +515,9 @@ object BuildSettings {
       ProblemFilters.exclude[ReversedMissingMethodProblem]("play.api.i18n.MessagesApi.langCookieMaxAge"),
       ProblemFilters.exclude[DirectMissingMethodProblem]("play.api.test.Helpers.stubMessagesApi"),
       ProblemFilters.exclude[DirectMissingMethodProblem]("play.api.test.StubMessagesFactory.stubMessagesApi"),
+      // Overload helpers so Symbol is not required
+      ProblemFilters.exclude[MissingTypesProblem]("views.html.helper.FieldElements$"),
+      ProblemFilters.exclude[DirectMissingMethodProblem]("play.api.test.StubMessagesFactory.stubMessagesApi"),
```
Comment (PromanSEW) [inline project/BuildSettings.scala]: Duplicate entry, see L517

**Comment by renatocaval** [comment]
@cookiee01, you need to add those exclusions as well:

```
ProblemFilters.exclude[IncompatibleSignatureProblem]("views.html.helper.FieldElements.unapply"),
ProblemFilters.exclude[DirectMissingMethodProblem]("views.html.helper.FieldElements.curried"),
ProblemFilters.exclude[DirectMissingMethodProblem]("views.html.helper.FieldElements.tupled"),
ProblemFilters.exclude[IncompatibleSignatureProblem]("views.html.helper.FieldElements.copy"),
ProblemFilters.exclude[IncompatibleSignatureProblem]("views.html.helper.FieldElements.apply"),
ProblemFilters.exclude[IncompatibleSignatureProblem]("views.html.helper.FieldElements.copy$default$4"),
ProblemFilters.exclude[IncompatibleSignatureProblem]("views.html.helper.FieldElements.this"),
ProblemFilters.exclude[IncompatibleSignatureProblem]("views.html.helper.FieldElements.unapply"),
ProblemFilters.exclude[IncompatibleSignatureProblem]("views.html.helper.FieldElements.apply"),
```

And remove the duplication mentioned by @PromanSEW

**Comment by mkurz** [comment]
Docs also needs update:
https://www.playframework.com/documentation/2.7.x/JavaFormHelpers#Form-template-helpers
https://www.playframework.com/documentation/2.7.x/ScalaForms#Form-template-helpers

**Comment by mkurz** [comment]
Please see https://github.com/playframework/playframework/issues/9447#issuecomment-529531637 and following. To move this pull request forward I think it makes sense to finally be able to deprecate twirl template arguments.

**Pull request closed without merging**